package/common_init/overlay/etc/init.d/S80endpoint_router_sbp

```diff
@@ -16,7 +16,7 @@ setup_permissions()
 
 starling_daemon_enabled=$(query_config --section experimental_flags --key starling_on_linux)
 
-if detect_piksi_ins; then
+if detect_piksi_ins_router && detect_piksi_ins; then
```

Comment:
Seems like we're calling `detect_piksi_ins_router` twice with this, won't `detect_piksi_ins` also call `detect_piksi_ins_router` ?

Reply:
good point.

Reply:
we are calling this twice, but it's an AND condition for the router, while it is in an or condition in detect_piksi_ins.  In order to use the smoothpose router, we need to be a real_piksi_ins and have config that requires smoothpose routing.

Reply:
In seeing that, I think I can make the logic less ugly and clearer though.  I will think for second and improve.

Reply:
I thought about it and I think this is still the clearest

Reply:
ok